### PR TITLE
Arithmetic slices problem in C++

### PR DESCRIPTION
**Pull request by ShreyashRoyzada**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes, I have.

### Description

Added the Arithmetic slices problem in C++ (Issue #2717). Updated the Documentation. Fixed a small typographical error present in the C++ Readme file.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

### CODE REVIEW

File: C-Plus-Plus/dp/Arithmetic_Slices.cpp [hunk 1]
```diff
@@ -0,0 +1,75 @@
+/*
+An integer array is called arithmetic if it consists of at least three elements and if the difference between any two consecutive elements is the same.
+
+For example, [1,3,5,7,9], [7,7,7,7], and [3,-1,-5,-9] are arithmetic sequences.
+Given an integer array nums, return the number of arithmetic subarrays of nums.
+
+A subarray is a contiguous subsequence of the array.
+
+*/
+
```
Comment (iamrajiv) [inline C-Plus-Plus/dp/Arithmetic_Slices.cpp:7]: @ShreyashRoyzada Remove these unwanted lines.

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@ShreyashRoyzada Add space and time complexity.

File: C-Plus-Plus/dp/Arithmetic_Slices.cpp (hunk 1 above)
Comment (ShreyashRoyzada) [reply C-Plus-Plus/dp/Arithmetic_Slices.cpp:7]: Added requested changes.

**Review by iamrajiv — APPROVED** [review]

File: C-Plus-Plus/dp/Arithmetic_Slices.cpp [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
```
Comment (HarshCasper) [inline C-Plus-Plus/dp/Arithmetic_Slices.cpp:7]: Add a description about the Code

File: C-Plus-Plus/dp/Arithmetic_Slices.cpp [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+#include<iostream>
+#include<vector>
+
+using namespace std;
+
+int numberOfArithmeticSlices(int A[],int size) {
```
Comment (HarshCasper) [inline C-Plus-Plus/dp/Arithmetic_Slices.cpp:14]: Add a function description

File: C-Plus-Plus/dp/Arithmetic_Slices.cpp [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+            if(res[i-1]>0)
+            {
+                res[i] = res[i-1]+1;
+            }
+            else
+            {
+            res[i]++;
+            }
+        }
+    }
```
Comment (HarshCasper) [inline C-Plus-Plus/dp/Arithmetic_Slices.cpp:38]: Put Comments describing the functionality

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by ShreyashRoyzada** [comment]
Added requested changes

File: C-Plus-Plus/dp/Arithmetic_Slices.cpp [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+using namespace std;
+
+//Function takes array and the size of the array as an input, it returns the number of arthemetic slices present in the input array//
+int numberOfArithmeticSlices(int A[],int size) {
+    if(size<3) return 0;
+    vector<int> res (size,0);
+    
+    /*Since as per the problem minimum size of the arthemetic slice can be 3, we start the loop from the 2nd element and check the previous elements.
+    We basically check for a continuous AP if the sequence continues, we change the res[i] value to res[1-1] + 1, if the AP breaks at any point we simply leave that index value at 0.
+    Thus restarting the sequence when another continuous AP is found.*/
```
Comment (HarshCasper) [inline C-Plus-Plus/dp/Arithmetic_Slices.cpp]: Make it compact to read and understand

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by atarax665** [comment]
@ShreyashRoyzada Are you still working on this?

**Comment by ShreyashRoyzada** [comment]
> @ShreyashRoyzada Are you still working on this?

Yes

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**